mangaki/mangaki/admin.py

```diff
@@ -1,20 +1,136 @@
 from django.contrib import admin
 from django.contrib.admin import helpers
-from django.template.response import TemplateResponse
-from django.db.models import Count
+from django.core.urlresolvers import reverse
 from django.db import transaction
+from django.db.models import Count
+from django.template.response import TemplateResponse
+from django.utils.html import format_html, format_html_join
 
 from mangaki.models import (
     Work, TaggedWork, WorkTitle, Genre, Track, Tag, Artist, Studio, Editor, Rating, Page,
     Suggestion, SearchIssue, Announcement, Recommendation, Pairing, Reference, Top, Ranking,
     Role, Staff, FAQTheme,
-    FAQEntry, ColdStartRating, Trope, Language
+    FAQEntry, ColdStartRating, Trope, Language, WorkCluster
 )
 from mangaki.utils.anidb import AniDB
 from mangaki.utils.db import get_potential_posters
 
 from collections import defaultdict
 from enum import Enum
+from datetime import datetime
+
+
+class MergeType(Enum):
+    INFO_ONLY = (0, 'black')
+    JUST_CONFIRM = (1, 'green')
+    CHOICE_REQUIRED = (2, 'red')
+
+    def __init__(self, priority, row_color):
+        self.priority = priority
+        self.row_color = row_color
+
+
+def overwrite_fields(final_work, request):
+    for field in request.POST.get('fields_to_choose').split(','):
+        if request.POST.get(field) != 'None':
+            setattr(final_work, field, request.POST.get(field))
+    final_work.save()
+
+
+def redirect_related_objects(works_to_merge, final_work):
+    for work_to_redirect in works_to_merge:
+        if work_to_redirect.id == final_work.id:
+            continue
+        for rating_to_redirect in work_to_redirect.rating_set.all():
+            try:
+                kept_rating = Rating.objects.get(work_id=final_work.id, user_id=rating_to_redirect.user_id)
```

Comment:
We should optimise the queries, maybe `kept_ratings = Rating.objects.filter(work_id=final_work.id, user_id__in=all_user_ids)` and iterate over it.